File: ClientINes/src/app/models/dtos/item.dto.ts
import { ItemStatus } from '../enums/item-status.enum';

export interface CreateCategoryDto {
  name: string;
}

export interface CreateItemDto {
  name: string;
  description?: string;
  categoryId: string;
  storageLocationId?: string;
  status: ItemStatus;
  purchaseDate?: string;
  purchasePrice?: number;
  estimatedValue?: number;
  currency: string;
  personName?: string;
  contactEmail?: string;
  expectedReturnDate?: string;
  sendNotification?: boolean;
  mainPhotoName?: string;
}

export interface UpdateItemDto extends Partial<CreateItemDto> {}

File: ServerINes/Models/DTOs/Item/ItemDto.cs
using INest.Models.Enums;

namespace INest.Models.DTOs.Item
{
    public class ItemDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Description { get; set; }

        public ItemStatus Status { get; set; }

        public decimal? EstimatedValue { get; set; }
        public string Currency { get; set; } = "USD";

        public string? PhotoUrl { get; set; }

        public Guid? StorageLocationId { get; set; }
        public string? StorageLocationName { get; set; }

        public Guid CategoryId { get; set; }
        public string CategoryName { get; set; } = null!;

        public bool IsOverdue { get; set; }
    }
}
